donate/payments/tasks.py

```diff
@@ -206,27 +205,26 @@ def process_charge_succeeded(self, event):
         subscription_id = charge.invoice.subscription
 
         try:
-            subscription = stripe.Subscription.retrieve(subscription_id)
+            subscription = stripe.Subscription.retrieve(subscription_id, expand=['customer'])
         except stripe.error.StripeError as e:
             logger.error(f'Error fetching Stripe Subscription: {e._message}', exc_info=True)
             return
 
         metadata = subscription.metadata
-        update_data = {'metadata': metadata}
         donation_url = None
 
-        if metadata.donation_url:
-            donation_url = metadata.donation_url
+        if 'donation_url' in metadata:
+            donation_url = metadata['donation_url']
 
-        if metadata.thunderbird:
-            update_data['description'] = 'Thunderbird monthly'
-        elif metadata.glassroomnyc:
-            update_data['description'] = 'glassroomnyc monthly'
+        if 'thunderbird' in metadata:
+            description = 'Thunderbird monthly'
+        elif 'glassroomnyc' in metadata:
+            description = 'glassroomnyc monthly'
         else:
-            update_data['description'] = 'Mozilla Foundation Monthly Donation'
+            description = 'Mozilla Foundation Monthly Donation'
 
         try:
-            stripe.Charge.modify(charge_id, update_data)
```

Comment:
This is what happens when you assume the signature of update methods is the same across languages (in this case, stripe-node VS stripe-python